transcom/mymove
# INT-B-21568 Display UB Preapproved Service Items

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1024473&RoomContext=TeamRoom%3A1007470&concept=TeamRoom)

## Summary
This story is to display the Preapproved Service Items for International UB shipments on the TOO Approval screen. Currently the domestic preapproved services are hard coded in the page. However for international we are pulling the data from re_service_items and using the shipment type, market code, auto approved data to determined which service items should display to the user.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Turn on ENABLE_ALASKA and UNACCOMPANIED_BAGGAGE feature flags.
2. Login as a Customer.
3. Create a Move.
4. Choose Alaska for the new duty location.
5. Add the orders document.
6. Add a new UB shipment to the move
7. Enter all required info.
8. Submit the move (note the move code).
9. Login as a Services Counselor.
10. Search for the move you created in the previous steps.
11. Enter the required orders info.
12. Submit the move.
13. Login as a TOO.
14. Search for the move.
15. On the move details page check the box by the UB shipment and then click the Approve Select button.
16. You should see the 4 applicable service items. Note International UB, International UB pack, International UB unpack will always show. For Conus to Oconus shipments, International POE Fuel Surcharge will display. For Oconus to Conus shipments, International POD Fuel Surcharge will display.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
Conus to Oconus
![image](https://github.com/user-attachments/assets/4e7d3f46-e2ba-447f-b936-ac99a1edffd5)

Oconus to Conus
![image](https://github.com/user-attachments/assets/ff612463-d33e-436d-ab68-0bb6278d00e7)
